x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/QueryRuleCriteria.java

```diff
@@ -69,15 +70,25 @@ public QueryRuleCriteria(QueryRuleCriteriaType criteriaType, @Nullable String cr
 
     public QueryRuleCriteria(StreamInput in) throws IOException {
         this.criteriaType = in.readEnum(QueryRuleCriteriaType.class);
-        this.criteriaMetadata = in.readOptionalString();
-        this.criteriaValues = in.readOptionalList(StreamInput::readGenericValue);
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_041)) {
```

Comment:
I think it's good to create a constant to refer to this `TransportVersion`, such as `CRITERIA_METADATA_TRANSPORT_VERSION`, so we can refer to it from other parts of the code and make it less error-prone.

Reply:
Good suggestion, thanks!